model_analyzer/record/metrics_manager.py

```diff
@@ -559,9 +559,20 @@ def _get_gpu_inference_metrics(self):
         # Stop and destroy DCGM monitor
         gpu_records = self._gpu_monitor.stop_recording_metrics()
 
+        if not gpu_records and self._expected_gpu_records():
```

Comment:
It still feels like the gpu monitor should have a heartbeat check, instead of passing this responsibility up to the metrics manager.